migration/identity-migration/src/main/java/io/camunda/migration/identity/config/ManagementIdentityConnector.java

```diff
@@ -7,27 +7,104 @@
+    final var properties = identityMigrationProperties.getManagementIdentity();
+    final var configuration =
+        new IdentityConfiguration(
+            properties.getBaseUrl(),
+            properties.getIssuerBackendUrl(),
+            properties.getIssuerBackendUrl(),
+            properties.getClientId(),
+            properties.getClientSecret(),
+            properties.getAudience(),
+            properties.getIssuerType());
+    return new Identity(configuration);
+  }
+
+  @Bean
   public ManagementIdentityClient managementIdentityClient(
       final RestTemplateBuilder builder,
-      final IdentityMigrationProperties identityMigrationProperties) {
+      final IdentityMigrationProperties identityMigrationProperties,
+      final Identity identity) {
     final var properties = identityMigrationProperties.getManagementIdentity();
     final var restTemplate =
         builder
             .rootUri(properties.getBaseUrl())
-            .defaultHeader("Authorization", String.format("Bearer %s", properties.getM2mToken()))
+            .interceptors(new M2MTokenInterceptor(identity, properties.getAudience()))
+            .errorHandler(new CustomErrorHandler())
             .defaultHeader("Content-Type", "application/json")
             .defaultHeader("Accept", "application/json")
             .build();
     return new ManagementIdentityClient(
         restTemplate, identityMigrationProperties.getOrganizationId());
   }
+
+  public static final class M2MTokenInterceptor implements ClientHttpRequestInterceptor {
+
+    final Identity identity;
+    final String audience;
+
+    public M2MTokenInterceptor(final Identity identity, final String audience) {
+      this.identity = identity;
+      this.audience = audience;
+    }
+
+    @Override
+    public ClientHttpResponse intercept(
+        final HttpRequest request, final byte[] body, final ClientHttpRequestExecution execution)
+        throws IOException {
+      final String token = identity.authentication().requestToken(audience).getAccessToken();
+      final HttpHeaders headers = request.getHeaders();
+      headers.add(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+      return execution.execute(request, body);
+    }
+  }
+
+  public static final class CustomErrorHandler extends DefaultResponseErrorHandler {
+
+    @Override
+    public boolean hasError(final ClientHttpResponse response) throws IOException {
+      return super.hasError(response) && response.getStatusCode() != HttpStatus.NOT_FOUND;
+    }
+
+    @Override
+    public void handleError(final ClientHttpResponse response) throws IOException {
+      if (response.getStatusCode() == HttpStatus.NOT_FOUND) {
+        return;
+      }
+      super.handleError(response);
+    }
+
+    @Override
+    public byte[] getResponseBody(final ClientHttpResponse response) {
+      try {
+        if (response.getStatusCode() == HttpStatus.NOT_FOUND) {
+          return "[]".getBytes(StandardCharsets.UTF_8);
+        }
```

Comment:
:question: What's the scenario where this is necessary? If the connection to Identity is configured but some endpoints return 404, we probably don't want to silently ignore this, right?

Reply:
We can add a log here but then ignore, because it's possible that we run migration app with a version of identity that doesn't support migration and then migration stuck in never ending retries.

Reply:
I think my problem with this is that it is a very blunt tool. Ignoring all 404 feels too dangerous, I'm afraid that we will miss actual bugs with this.